src/JsonRpcProxy/Adaptor.js

```diff
@@ -0,0 +1,266 @@
+        format.formatEpochOfParams(req.params, 2);
+        await next();
+    }
+
+    async function getTransactionByBlockHashAndIndex(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        let txIndex = req.params[1];
+        req.params[1] = true;
+        await next();
+        if(!res || !res.result) return;
+        let index = Number(txIndex);
+        res.result = res.result.transactions[index];
+    }
+
+    async function getTransactionByBlockNumberAndIndex(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        let txIndex = req.params[1];
+        req.params[1] = true;
+        await next();
+        if(!res || !res.result) return;
+        let index = Number(txIndex);
+        res.result = res.result.transactions[index];
+    }
+
+    async function getTransactionByHash(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        await next();
+        if(!res || !res.result) return;
+        format.formatTransaction(res.result);
+        const block = await proxy.cfx.getBlockByHash(res.result.blockHash);
+        res.result.blockNumber = util.numToHex(block.epochNumber);
+    }
+
+    async function getTransactionCount(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        req.params[0] = format.formatAddress(req.params[0], proxy.networkId);
+        format.formatEpochOfParams(req.params, 1);
+        await next();
+    }
+
+    async function getTransactionReceipt(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        await next();
+        if (!res || !res.result) return;
+        let txReceipt = res.result;
+        if (txReceipt.contractCreated) {
+            txReceipt.contractCreated = format.formatHexAddress(txReceipt.contractCreated);
+        }
+        txReceipt.from = format.formatHexAddress(txReceipt.from);
+        txReceipt.to = format.formatHexAddress(txReceipt.to);
+        if (txReceipt.logs) {
+            txReceipt.logs.forEach(
+            l => (l.address = format.formatHexAddress(l.address))
+            );
+        }
+
+        txReceipt.contractAddress = txReceipt.contractCreated;
+        txReceipt.blockNumber = txReceipt.epochNumber;
+        txReceipt.transactionIndex = txReceipt.index;
+        txReceipt.status = Number.parseInt(txReceipt.outcomeStatus)
+            ? "0x0"
+            : "0x1"; // conflux and eth status code is opposite
+        txReceipt.cumulativeGasUsed = txReceipt.gasUsed; // NOTE: this is an fake value
+        util.delKeys(txReceipt, [
+            "contractCreated",
+            "epochNumber",
+            "index",
+            "outcomeStatus",
+            "stateRoot"
+        ]);
+        return res;
+    }
+
+    async function sendTransaction(req, res, next) {
+        req.method = methodAdaptor(req.method, req.params);
```

Comment:
Some fields of an transaction parameter can be filled automatically for example: nonce, gas, gasPrice, chainId and so on

Reply:
fixed